conductor_api/src/conductor/base.rs

```diff
@@ -1230,6 +1277,187 @@ pub mod tests {
+    /// a decrepency btween DNA hashes.
+    fn test_check_dna_consistency_err() {
+        let _toml = format!(
+            r#"
+        [[agents]]
+        id = "test-agent-1"
+        name = "Holo Tester 1"
+        public_address = "{}"
+        keystore_file = "holo_tester1.key"
+
+        [[agents]]
+        id = "test-agent-2"
+        name = "Holo Tester 2"
+        public_address = "{}"
+        keystore_file = "holo_tester2.key"
+
+        [[agents]]
+        id = "test-agent-3"
+        name = "Holo Tester 3"
+        public_address = "{}"
+        keystore_file = "holo_tester3.key"
+
+        [[dnas]]
+        id = "test-dna"
+        file = "app_spec.dna.json"
+        hash = "QmQVLgFxUpd1ExVkBzvwASshpG6fmaJGxDEgf1cFf7S73a"
+
+        [[dnas]]
+        id = "bridge-callee"
+        file = "bridge/callee.dna"
+        # This should fail
+        # hash = "QmZAQkpkXhfRcSgBJX4NYyqWCyMnkvuF7X2RkPgqihGMrR"
+        hash = "QmYRM4rh8zmSLaxyShYtv9PBDdQkXuyPieJTZ1e5GZqeeh"
+
+        [[dnas]]
+        id = "bridge-caller"
+        file = "bridge/caller.dna"
+        hash = "QmYRM4rh8zmSLaxyShYtv9PBDdQkXuyPieJTZ1e5GZqeeh"
+
+        [[instances]]
+        id = "test-instance-1"
+        dna = "bridge-callee"
+        agent = "test-agent-1"
+
+            [instances.storage]
+            type = "memory"
+
+        [[instances]]
+        id = "test-instance-2"
+        dna = "test-dna"
+        agent = "test-agent-2"
+
+            [instances.storage]
+            type = "memory"
+
+        [[instances]]
+        id = "bridge-caller"
+        dna = "bridge-caller"
+        agent = "test-agent-3"
+
+            [instances.storage]
+            type = "memory"
+        "#,
+            test_keybundle(1).get_id(),
+            test_keybundle(2).get_id(),
+            test_keybundle(3).get_id(),
+        );
+        // let config = load_configuration::<Configuration>(&_toml).unwrap();
+        // let mut conductor = Conductor::from_config(config.clone());
+        // conductor.dna_loader = test_dna_loader();
+        // conductor.key_loader = test_key_loader();
+        // assert_eq!(
+        //     conductor.boot_from_config(),
+        //     Err(String::from("Error while trying to create instance \"test-instance-1\": DNA hash consistency fail: \
+        //                      There is a discrepancy between the DNA hash provided in the \
+        //                      Conductor configuration file and the hash computed from the \
+        //                      loaded instance of the Conductor. Please check the logs for additional information")),
+        //     "Conductor failed to boot from config"
+        // );
+
+        let a = HashString::from("QmYRM4rh8zmSLaxyShYtv9PBDdQkXuyPieJTZ1e5GZqeeh");
+        let b = HashString::from("QmZAQkpkXhfRcSgBJX4NYyqWCyMnkvuF7X2RkPgqihGMrR");
+
+        assert_eq!(
+            Conductor::check_dna_consistency(&a, &b),
+            Err(HolochainInstanceError::InternalFailure(
```

Comment:
What do we prefer here?
- `HolochainInstanceError::InternalFailure`
or
- `HolochainError::DnaHashMismatch`